internal/schedule/penalize/penalize.go

```diff
@@ -0,0 +1,39 @@
+package penalize
+
+import (
+	"github.com/NodeFactoryIo/vedran/internal/active"
+	"github.com/NodeFactoryIo/vedran/internal/models"
+	"github.com/NodeFactoryIo/vedran/internal/repositories"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+const MaxCooldownForPenalizedNode = 17 * time.Hour
+
+var afterFunc = time.AfterFunc
+
+func ScheduleCheckForPenalizedNode(node models.Node, repositories repositories.Repos) {
+	afterFunc(time.Duration(node.Cooldown), func() {
+		isActive, err := active.CheckIfNodeActive(node, &repositories)
+		if err != nil {
+			log.Errorf("Unable to check if node %s active, because of %v", node.ID, err)
+		}
+
+		if isActive {
+			repositories.NodeRepo.AddNodeToActive(node)
+		} else {
+			newCooldown := node.Cooldown * 2
```

Comment:
Can this be some repo method to clean this up a bit and so we have it if we want to increase cooldown somewhere else?